spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestBaseReader.java

```diff
@@ -47,14 +49,13 @@
 import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 import org.apache.iceberg.spark.data.RandomData;
 import org.apache.iceberg.types.Types;
-import org.junit.Assert;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
 
 public class TestBaseReader {
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir
+  public Path temp;
```

Comment:
with JUnit5 these can all be `private` now. Please also adjust accordingly in other files

Reply:
updated

Reply:
I'm now getting build errors on private access.
Example:
```
error: temp has private access in TestIcebergSourceTablesBase
    this.tableDir = temp.toFile();
```

In [this PR](https://github.com/apache/iceberg/pull/9129#discussion_r1403012159), you mentioned using `protected` instead of `public`. Can I do that instead?

Reply:
if it's used in a subclass then `protected` makes sense, otherwise it should be `private`